### PR TITLE
Add MAFFT package

### PR DESCRIPTION
**Pull request by hmacdope**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hmacdope** [comment]
@mikemhenry would you be willing to be a maintainer on this package?

**Comment by hmacdope** [comment]
@conda-forge/bioconda-recipes, pinging :smile:

**Comment by mikemhenry** [comment]
I am willing to maintain this package

**Comment by hmacdope** [comment]
@conda-forge/help-c-cpp I think this should be ready for review!

**Review by zklaus — COMMENTED** [review]
Thanks, @hmacdope!

Lgtm. Did you already hear from bioconda?

**Comment by hmacdope** [comment]
Thanks @zklaus! No word from them.

**Comment by zklaus** [comment]
@conda-forge/bioconda, this PR aims to move a bioconda recipe to conda-forge. Would you be able to comment?

**Comment by hmacdope** [comment]
@conda-forge/help-c-cpp could I snag a review from yall? RE bioconda @zklaus, shall we say they have been notified and we can move forward? We didn't hear back from them on a similar PR of mine.

**Comment by bgruening** [comment]
Bioconda is in general not opposed to move this recipe, however a reason that justifies this move would be nice. In addition there is the problem that the bioconda version is packaging a more recent version afaics. Having only an older version here will cause problems for users with strict channel priority.

**Comment by hmacdope** [comment]
@bgruening I am happy to package the more recent version if you like. 

As for justification, see below, happy to discuss. 

> @bgruening we want to have this package (asapdiscovery: https://github.com/choderalab/asapdiscovery) be all conda-forge installable, so that this package can be deployed with conda-forge itself.

> I suppose in this case it isn't strictly necessary as we don't import or link to MAFFT and instead run as an executable, but the 
 idea remains that you should be able to install deps for a conda-forge package strictly from the conda-forge channel.

**Comment by hmacdope** [comment]
@bgruening would you still be open to migrating? I have updated to the latest version and  have added you as a maintainer.

**Comment by hmacdope** [comment]
The windows packages (and potential ARM packages) are at least a solid value add from migrating.

**Comment by kcpevey** [comment]
@bgruening any thoughts on the justification?

**Comment by bgruening** [comment]
I don't need to be a maintainer. I'm raising concerns about maintenance of mafft. I know bioconda is maintaining it and I'm unsure if it will be here.  Now coming up with some random justifications just to get it in also leaves a bad taste ;-) especially that bioconda is supporting ARM and the containers that bioconda is generating can be used on a variety of other architectures.

That said please go ahead, I will not block it any longer.
Thanks for your work!

**Review by bgruening — APPROVED** [review]

File: recipes/mafft/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set name = "mafft" %}
```
Comment (jaimergp) [inline recipes/mafft/meta.yaml:1]: Can you add a comment somewhere saying that the recipe comes bioconda?

Comment (hmacdope) [reply recipes/mafft/meta.yaml:1]: Done

File: recipes/mafft/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# recipie from bioconda https://github.com/bioconda/bioconda-recipes/blob/master/recipes/mafft/meta.yaml
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://gitlab.com/sysimm/mafft/-/archive/v{{ version }}/mafft-v{{ version }}.zip
+  sha256: "fbd94516539c3401243d80f0c33ee247362120b0efc92b31260b2d5db484b2a6"
```
Comment (jaimergp) [inline recipes/mafft/meta.yaml:12]:
```suggestion
  sha256: fbd94516539c3401243d80f0c33ee247362120b0efc92b31260b2d5db484b2a6
```

File: recipes/mafft/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+build:
+  skip: true  # [win or py<39]
```
Comment (jaimergp) [inline recipes/mafft/meta.yaml:15]:
```suggestion
  skip: true  # [win]
```

**Comment by corneliusroemer** [comment]
Please add back linux-aarch64 support. If you want the package to be in conda-forge you should not reduce the architecture support as you did here, when migrating.

**Pull request merged**